sql: update br docs about some sql options (#17416)
pingcap/docs: sql-statements/sql-statement-restore.md

    "RATE_LIMIT" '='? LengthNum "MB" '/' "SECOND"
|   "CONCURRENCY" '='? LengthNum
|   "CHECKSUM" '='? Boolean

> |   "CHECKSUM_CONCURRENCY '='? LengthNum

Reviewer: ```suggestion
|   "CHECKSUM_CONCURRENCY" '='? LengthNum
```

Use `RATE_LIMIT` to limit the average download speed per TiKV node to reduce network bandwidth.

> Before restore is completed, `RESTORE` would perform a checksum against the data from the archive to verify correctness. 

Reviewer: ```suggestion
Before the restore is completed, `RESTORE` would perform a checksum against the data in the backup files to verify correctness.
```

> Use `CHECKSUM` to disable the check if you are confident that this is unnecessary.

Reviewer: ```suggestion
You can use the `CHECKSUM` parameter to disable the check if you are confident that this verification is unnecessary.
```

Use `RATE_LIMIT` to limit the average download speed per TiKV node to reduce network bandwidth.

Before restore is completed, `RESTORE` would perform a checksum against the data from the archive to verify correctness. 

Use `CHECKSUM` to disable the check if you are confident that this is unnecessary.

> Use `CHECKSUM_CONCURRENCY` to controll the concurrency of checksumming in one table (default 4)

Reviewer: ```suggestion
Use `CHECKSUM_CONCURRENCY` to control the concurrency of checksum verification in a single table, and the default value is 4.
```
